<?xml version="1.0"?>
<launch>

	<!-- Map Server -->
    <node name="map_server" pkg="map_server" type="map_server" args="$(find map_server)/test/testmap.yaml">
        <param name="frame_id" value="/map"/>
    </node>

    <!-- rViz -->
    <node pkg="rviz" type="rviz" name="rviz" />

    <!-- URDF Models -->
    <param name="robot_description" textfile="$(find car_model)/urdf/car.urdf" />
    <param name="pedestrian_model" textfile="$(find car_model)/urdf/pedestrian.urdf" />
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />

    <!-- Car and Pedestrian Node / Model -->
    <node name="car" pkg="car_model" type="Car_Node.py" output="screen"/>
    <!-- <node name="Pedestrian_Model" pkg="car_model" type="pedestrian_model.py" output="screen"/> -->

    <!-- Global and Local Planner -->
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <rosparam file="$(find car_model)/params/costmap_common.yaml" command="load" ns="global_costmap"/>
        <rosparam file="$(find car_model)/params/costmap_common.yaml" command="load" ns="local_costmap"/>
        <rosparam file="$(find car_model)/params/global_costmap.yaml" command="load"/>
        <rosparam file="$(find car_model)/params/local_costmap.yaml" command="load"/>
        <rosparam file="$(find car_model)/params/teb_local_planner.yaml" command="load"/>

        <param name="base_global_planner" value="global_planner/GlobalPlanner"/> 
        <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
    </node>

    <!-- Goal Node for visualizing Goal Markers -->
    <node name="goal" pkg="car_model" type="goal.py" output="screen"/>

    <!-- Planner for the pedestrian -->
    <node name="pedestrian_planner" pkg="car_model" type="pedestrian_planner.py" output="screen"/>

    <!-- Node for calculating the Observation Probability for DESPOT -->
    <node name="prob_obs_pos" pkg="car_model" type="prob_obs_pos.py" output="screen"/>

    <!-- Interactivae Test Marker -->
    <node name="interactive_marker" pkg="car_model" type="test_obstacles.py" output="screen"/>

    <!-- Publishes the voxel grid to rviz for display -->
    <node pkg="costmap_2d" type="costmap_2d_node" name="costmap_2d">
        <!-- <remap from="voxel_grid" to="costmap/voxel_grid"/> -->
    </node>


</launch>
